test/modules/standard/memory/countMemory/countMemory.prediff

```diff
@@ -14,12 +15,20 @@ if (-r $memfile) {
     }
   }
 } else {
-    $platform = `$ENV{CHPL_HOME}/util/chplenv/chpl_platform.py --target`; chomp($platform);
     if ($platform eq "darwin") {
         $memFreeInBytes = `sysctl -n hw.memsize`; chomp($memFreeInBytes);
     }
 }
 
+# cygwin32 running on a 64 bit windows can have more than 4GB reported by
+# /proc/meminfo, but only ~4GB is actually accessible
+if ($platform eq "cygwin32") {
+    $maxMemInBytes = 0xffffffff;
+    if ($memFreeInBytes > $maxMemInBytes) {
+        $memFreeInBytes = $maxMemInBytes;
+    }
```

Comment:
I actually don't feel uneasy about this at all, I'm really happy with this solution. /proc/meminfo reports the total amount of memory, not the total amount of addressable memory. cygwin gets the values for /proc/meminfo from windows, and there could very well be more than 4GB of memory, it just can't address into it.

The "problem" is really that we assumed that /proc/meminfo represents how much memory is within our virtual memory range, but that's just not true for this situation, so we have to adjust how we compute the total virtual memory.

This shouldn't be expanded to include other *32 platforms. Any other 32 bit machine's total memory is limited by an upper bound of how much is addressable. i.e. if you have a machine with 32 GB of ram, but it's running linux32, it'll only report having 4GB total.  cygwin32 is special because it's gets the total amount from a 64 bit windows.